### DISCUSSION TITLE
cannot import name 'TaskType' from 'anomalib.data'

### DISCUSSION THREAD

**Discussion by monkeycc** [open-edge-platform/anomalib#1570] [Q&A] [answered]
https://anomalib.readthedocs.io/en/v1/markdown/guides/reference/data/image/folder.html

```python
folder_datamodule = Folder(
    root=dataset_root,
    normal_dir="good",
    abnormal_dir="crack",
    task=TaskType.SEGMENTATION,
    mask_dir=dataset_root / "mask" / "crack",
    image_size=256,
    normalization=InputNormalizationMethod.NONE,
)
folder_datamodule.setup()

```
Folder ?
`from anomalib.data.image import Folder `
??

TaskType ？
`from anomalib.data import TaskType`
？？

from anomalib.data import TaskType
ImportError: cannot import name 'TaskType' from 'anomalib.data' (E:\anomalib\src\anomalib\data\__init__.py)

anomalib                  1.0.0.dev0   E:\anomalib

```python

folder_dataset_classification_train = FolderDataset(
    normal_dir=dataset_root / "good",
    abnormal_dir=dataset_root / "crack",
    split="train",
    transform=transform,
    task=TaskType.CLASSIFICATION,
)

```

FolderDataset ??
`from anomalib.data.image.folder.FolderDataset  import FolderDataset`
No module named 'anomalib.data.image.folder.FolderDataset'; 'anomalib.data.image.folder' is not a package

**Comment by samet-akcay** [comment, marked as answer]
@monkeycc, can you confirm whether you checked out the [v1](https://github.com/openvinotoolkit/anomalib/tree/v1), from which you install anomalib?

If you install it properly you should have the following imports:


## Folder
```python
from anomalib.data import Folder
# or
from anomalib.data.image.folder import FolderDataset
```

## TaskType
```python
from anomalib.utils.types import TaskType
```

> **NOTE**: `TaskType has been moved to `anomalib.utils.types`. The main reason is that it is also used by some other components within anomalib, so we had to place it to a more generic location

## FolderDataset
```python
from anomalib.data.image.folder import FolderDataset
```

Let me know if you still have any questions. Good luck!
